src/main/java/spl/EchoThread.java

```diff
@@ -0,0 +1,51 @@
+package spl;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.Socket;
+import java.util.ArrayList;
+
+public class EchoThread extends Thread {
+    protected Socket socket;
+    protected ArrayList<Socket> others;
+
+    public EchoThread(Socket clientSocket, ArrayList<Socket> others) {
+        this.socket = clientSocket;
+        this.others = others;
+    }
+
+    public void run() {
+        InputStream inp = null;
+        BufferedReader brinp = null;
+        DataOutputStream out = null;
+        try {
+            inp = socket.getInputStream();
+            brinp = new BufferedReader(new InputStreamReader(inp));
+            
+        } catch (IOException e) {
+            return;
+        }
+        String line;
+        while (true) {
+            try {
+                line = brinp.readLine();
+                if ((line == null) || line.equalsIgnoreCase("QUIT")) {
```

Comment:
leesbaarder als je een methode met naam hieraan geeft (we moeten hier vgm nog lang mee verder)

Reply:
eens heb er een functie voor gemaakt